test-clients/src/main/java/com/hedera/services/bdd/spec/queries/meta/HapiGetTxnRecord.java

```diff
@@ -288,6 +321,92 @@ protected void assertExpectationsGiven(HapiApiSpec spec) throws Throwable {
 					0,
 					actualRecord.getTokenTransferListsCount());
 		}
+		if (validateAccountAmountsInTransferList) {
+			final var accountAmounts = actualRecord.getTransferList().getAccountAmountsList();
+			accountAmountsToValidate.forEach(pair ->
+					validateAccountAmount(spec.registry().getAccountID(pair.getLeft()), pair.getRight(), accountAmounts));
+		}
+		if (validateTokenAmountsInTokenTransferList) {
+			final var tokenTransferLists = actualRecord.getTokenTransferListsList();
+			tokenAmountsToValidate.forEach(triple ->
+					validateTokenTransferList(
+							spec.registry().getTokenID(triple.getLeft()),
+							spec.registry().getAccountID(triple.getMiddle()),
+							triple.getRight(),
+							tokenTransferLists));
+		}
+		if (validateEntitiesInAssessedCustomFees) {
+			final var actualAssessedCustomFees = actualRecord.getAssessedCustomFeesList();
+			assessedCustomFeesToValidate.forEach(triple ->
+					validateAssessedCustomFees(
+							!triple.getLeft().equals(HBAR_TOKEN_SENTINEL) ? spec.registry().getTokenID(triple.getLeft()) : null,
+							spec.registry().getAccountID(triple.getMiddle()),
+							triple.getRight(),
+							actualAssessedCustomFees
+					));
+		}
+	}
+
+	private void validateAssessedCustomFees(final TokenID tokenID, final AccountID accountID, final long amount,
+			final List<AssessedCustomFee> assessedCustomFees) {
+		boolean found = false;
+		for (var acf : assessedCustomFees) {
+			if (acf.getAmount() == amount &&
+					acf.getFeeCollectorAccountId().equals(accountID) &&
+					(!acf.hasTokenId() || acf.getTokenId().equals(tokenID))) {
+				found = true;
+				break;
+			}
+		}
+		assertTrue("Cannot find TokenID : " + tokenID +
+				" AccountID : " + accountID +
+				" and amount : " + amount + " in assessed_custom_fees of the txnRecord", found);
+	}
+
+	private void validateTokenTransferList(final TokenID tokenID, final AccountID accountID, final long amount,
+			final List<TokenTransferList> tokenTransferLists) {
+		boolean found = false;
+
+		for (final var ttl : tokenTransferLists) {
+			if (ttl.getToken().equals(tokenID)) {
+				final var accountAmounts = ttl.getTransfersList();
+				for (var aa : accountAmounts) {
+					if( aa.getAmount() == amount &&
+							aa.getAccountID().equals(accountID)) {
+						found = true;
+						break;
+					}
+				}
+				final var nftTransferList = ttl.getNftTransfersList();
```

Comment:
Here, before getting into the next `for` loop of nftTransferList, I think a check of the `found` value is necessary. Only get into the nftTransferList loop if `found` is `false`. Or have another `if(found) { break; }` here to break the outer loop immediately here.   

```

Reply:
@ljianghedera we will only have either `TransfersList` or `NftTransfersList` in a TokenTransferList. So one of them will be empty.

Reply:
nit: then I would suggest to put the two `for` loops in a `if...else...` structure (probably refactored into a separate function to avoid code complexity) to make it semantically clear that only one loop will happen in any conditions.